code/game/mecha/equipment/weapons/weapons.dm

```diff
@@ -12,15 +13,44 @@
 			return 1
 	return 0
 
+/obj/item/mecha_parts/mecha_equipment/weapon/become_damaged()
+	if(!damaged)
+		damaged = 1
+		desc += "/nIt doesn't look to be in the best shape."
+	return ..()
+
+/obj/item/mecha_parts/mecha_equipment/weapon/proc/get_inaccuracy(var/atom/target, var/spread, var/obj/mecha/chassis)
```

Comment:
I'm not sure how to put both the mech versions and normal gun versions together in a way that'll work